### DISCUSSION TITLE
Switching to NFS for Installation sources?

### DISCUSSION THREAD

**Discussion by stoeps13** [HCL-TECH-SOFTWARE/connections-automation#25] [Ideas]
Hi Dejan,

I read through the quickstart.md and thought about my first steps with Ansible. ;)

One point for discussion. I would suggest to move the installation sources to a folder on the nfs server (and extract it there), so we can save some diskspace. In the moment on each WebSphere Node the Websphere packages and fixpacks are downloaded and extracted. So each nodes has an overhead of 10-20GB.
Using NFS could be done with an initial task which checks the source packages and extract them. Then all inventory server can mount ie. /opt/install and use the packages from there.

Just an idea.

Regards
Christoph

**Comment by rboccadoro** [comment]
I would have it as an option, not the default. Not all customers use nfs so an option to have the sources locally should still exist

**Reply by dejo1307** [reply]
Agreed, that is on the same line what I am doing now. It is currently being tested, trying to find the least disruptive option.

**Reply by dejo1307** [reply]
Hey @rboccadoro it is merged now, sorry for waiting, took us some time to properly test it.

This line - https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/5a847494ef7a86446bef926b71270a634104f908/environments/examples/cnx7/connections_7_with_component_pack/connections#L80 - or better to say this param, when uncommented, will tell Connections not to use NFS, so Connections will not try to mount anything. Note that this is all good and fine (and you probably know it better then I but just writing it in case someone else reads it) but NFS is required if you are going to set up Docs or if you are going to use anything but single node installation. By default, NFS is enabled, and we removed NFS from setup-connectinos-only.yml playbook (because if it says also Connections only should be Connections only) but in case you still want to add NFS there is a playbook for that as well.

**Comment by stoeps13** [comment]
Even then we can just create a NFS Share on one of the used servers and share it with the others. Just to save space and share configs without being dependent on SSH or Rsync.

**Comment by dejo1307** [comment]
Sorry, I somehow missed this thread. 

Actually, to make it more simple - and I started working on this as PoC - making install sources available from file system in general (in that case will work with NFS of course). Will definitely help also to those who already downloaded packages, or in cases when you are setting things locally in your VMs and stuff.

**Reply by dejo1307** [reply]
I see I answered above, but I had a different topic in my head :) But still, I am working on this, AND NFS being not mandatory is also being tested as a separate topic.

**Comment by rboccadoro** [comment]
Hi Dejan

Actually you can install Docs without NFS. Just create two directories ( I 
usually use /opt/HCL/DocsShare and /opt/HCL/ViewerShare ) and when you 
install Docs using scripts just point to them in the cfg.properties files 

RoB

**Comment by dejo1307** [comment]
Forgot to write update, but NFS can now be used (or simply to say local file system) as installation source. Instead of putting "http://..." simply the path to the files can be put as the value in https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/355d0ff299963b3dcce41c2053b43eae1453581f/environments/examples/cnx7/connections_7_with_component_pack/connections#L48-L76

I will update the documentation tomorrow properly with this, but it is on main already, and I tested it all around.
